src/domain/core/src/repos/dataset_repository.rs

```diff
@@ -52,6 +52,11 @@ pub trait DatasetRepository: DatasetRegistry + Sync + Send {
 
     fn get_all_datasets(&self) -> DatasetHandleStream<'_>;
 
+    fn get_all_datasets_filtered(
+        &self,
+        dataset_ref_pattern: DatasetRefPattern,
+    ) -> DatasetHandleStream<'_>;
```

Comment:
I think the filtering algorithm does not depend on repository kind - so it should not be expanded in implementatino structs anyhow differently.

It could be a static method with default implementation, or even a global utility function, which takes a stream of dataset handles + pattern, and filters them depending if they satisfy the pattern.